packages/ember-metal/lib/meta.js

```diff
@@ -183,27 +183,55 @@ export class Meta {
     return this[key] || (this[key] = Object.create(null));
   }
 
-  _getInherited(key) {
+  _forEachInherited() {
     let pointer = this;
+    let keys = arguments;
+    let len = keys.length - 1;
+    let callback = arguments[len];
+    let value, key, i;
+    let seen = Object.create(null);
+
     while (pointer !== undefined) {
-      let map = pointer[key];
-      if (map !== undefined) {
-        return map;
+      value = pointer;
+      i = 0;
+      for (; i < len; i++) {
+        key = keys[i];
+        value = value[key];
+        if (value === undefined) {
+          break;
+        }
+      }
+
+      if (i === len) {
+        for (let innerKey in value) {
+          if (seen[innerKey] === undefined) {
+            seen[innerKey] = true;
+            callback(innerKey, value[innerKey]);
+          }
+        }
       }
+
       pointer = pointer.parent;
     }
   }
 
-  _findInherited(key, subkey) {
+  _findInherited() {
```

Comment:
accepts list of keys, when `obj.prop1.prop2.prop3` `_findInherited('obj','prop1','prop2','prop3')` will find first `prop3` property that has `not undefined` value and returns it